WordPress/src/test/java/org/wordpress/android/ui/jetpack/backup/download/BackupDownloadViewModelTest.kt

```diff
@@ -1,30 +1,160 @@
 
+    private val rewindId = "rewindId"
+    private val downloadId = 100L
+    private val published = Date(1609690147756)
+    private val url = "url"
+
+    private val backupDownloadState = BackupDownloadState(
+            activityId = "activityId",
+            rewindId = "rewindId",
+            downloadId = 100L,
+            siteId = 200L,
+            url = null,
+            published = Date(1609690147756)
+    )
+
     @Before
     fun setUp() {
+        whenever(wizardManager.navigatorLiveData).thenReturn(wizardManagerNavigatorLiveData)
+        whenever(wizardManager.showNextStep()).then {
+            wizardManagerNavigatorLiveData.value = backupDownloadStep
+            Unit
+        }
         viewModel = BackupDownloadViewModel(wizardManager)
     }
 
+    @Test
+    fun `when view model is started, process moves to next step`() {
+        viewModel.start(null)
+
+        verify(wizardManager).showNextStep()
+    }
+
+    @Test
+    fun `when details is finished, process moves to next step`() {
+        viewModel.start(null)
+        // need to clear invocations because nextStep is called on start
+        clearInvocations(wizardManager)
+
+        viewModel.onBackupDownloadDetailsFinished(rewindId, downloadId, published)
+        verify(wizardManager).showNextStep()
+    }
+
+    @Test
+    fun `when progress is finished, process moves to next step`() {
+        viewModel.start(null)
+        clearInvocations(wizardManager)
+
+        viewModel.onBackupDownloadProgressFinished(url)
+        verify(wizardManager).showNextStep()
+    }
+
+    @Test
+    fun `when details is finished, state is updated properly`() {
+        val navigationTargets = initObservers().navigationTargets
+
+        viewModel.start(null)
+        clearInvocations(wizardManager)
+        whenever(wizardManager.showNextStep()).then {
+            wizardManagerNavigatorLiveData.value = BackupDownloadStep.PROGRESS
+            Unit
+        }
+
+        viewModel.onBackupDownloadDetailsFinished(rewindId, downloadId, published)
+
+        assertThat(navigationTargets.last().wizardState.rewindId).isEqualTo(rewindId)
+        assertThat(navigationTargets.last().wizardState.downloadId).isEqualTo(downloadId)
+        assertThat(navigationTargets.last().wizardState.published).isEqualTo(published)
+    }
+
+    @Test
+    fun `when progress is finished, state is updated properly`() {
+        val navigationTargets = initObservers().navigationTargets
+
+        viewModel.start(null)
+        clearInvocations(wizardManager)
+        whenever(wizardManager.showNextStep()).then {
+            wizardManagerNavigatorLiveData.value = BackupDownloadStep.COMPLETE
+            Unit
+        }
+
+        viewModel.onBackupDownloadProgressFinished(url)
+
+        assertThat(navigationTargets.last().wizardState.url).isEqualTo(url)
+    }
+
+    @Test
+    fun `when onBackPressed while in details step, invokes wizard finished with cancel`() {
```

Comment:
1) Question (❓): I am not sure if you want to utilise the `given, when, then` approach, but if you do want that, this test scenario could have been written as such: `given in detail step, when onBackPressed, then invokes wizard finished with cancel`

PS: The same apply to all tests. For example, the `then` wording is missing on all of them. Again, not sure if that is what you want. If not, please ignore this comment.